### ISSUE TITLE
iGPU frequency locked at .55GHz?

### ISSUE THREAD

**Issue by coreygreenberg** [ameeno/Dell-E7440-Hackintosh#22]
Might only be cosmetic but Intel Power Gadget shows the HD 4400 iGPU to be locked at .55GHz, even at idle. Everything seems to be working fine but I wonder if maybe the included DSDT and/or SSDT files are somehow overlocking the iGPU? I'm running this package on a Dell E7240. Thanks in advance for any feedback.

**Comment by ameeno** [comment]
Hello mate, thanks for your message.

To explore further I need to know:

Is that on battery or with power connected?


On my settings
I prefer to reduce the power usage by disabling turbo boost on battery with
this build.

If you remove the disableturboboost kext and reboot this action will stop
(remember to clear cache)



Thee graphics are managed by the whatevergreen kext and there may be some
restrictions to enable stability.


I feel As long as the graphics are good enough for me to do 3 monitors at
high resolution with no lag I am fine.

But really I wouldn't game on a hackintosh or even use a hackintosh laptop
for video editing. It is seriously underpowered and more native software is
available on windows (except final cut Pro)

I cannot fathom a reason why you would need a faster clock on your GPU and
the speed restricted clock leads to stability, better battery life and less
heat generated.

Actually:
Most people struggle on their hackintosh builds to get the heat down and
the battery life up, so actually a clocked down GPU to me sounds like a
very good build in deed!




Now I have not actually tested a MacBook Haswell pro for comparison, but a
frequency lock may be a native applr feature. Or perhaps it's just the
sensor readings and not the actual clock speed. Perhaps the gfx sensor info
provided by the kexts is not the true frequency?

To test that:

Have you run system benchmarks on gfx and noticed any numbers very
different on windows or Linux with the same laptop?

Benchmark scores wildly different across os's?

Just some ideas and tips for you to consider.

Please let me know your thoughts and how you plan to proceed.

**Comment by coreygreenberg** [comment]
Mate! I cannot thank you enough for sharing your Dell files. I have 2 E7240s that I’ve never been able to really get Hackintoshed all the way to usability until I found your Git and got both my Dells to fully working Mojave machines. You are a genius!

I’m with you on disabling turbo, I did that in BIOS anyway. No real benefit on such lightweight hardware. 

I think maybe I wasn’t as clear about the locked .55GHz iGPU as I should have been. .55GHz seems to be the baseline as reported by Intel Power Gadget. When I push the HD-4400 with Geekbench graphics benchmarks, Intel Power Gadget shows GPU frequency scaling up as you would expect. But instead of the range being 0 as the baseline, it never dips below .55GHz whether on battery or AC. Oddly enough if I look at Activity Monitor, it shows the GPU scaling from 0 to whatever. But Intel Power Gadget seems to see the GPU idling at .55GHz, which is a pretty significant load when the system’s at idle. 

This may be entirely cosmetic and Intel’s app is misreading the GPU while Apple’s Activity Monitor is showing the actual state. I just wanted to run it by you in case you were seeing the same thing.

Again, thanks for the files! You are the only one who’s managed to crack the code on these Dells.

Best regards,
Corey Greenberg

**Comment by ameeno** [comment]
No probs pal.

It's not all me its I have simply compiled packs from others and made
changes to it.


Now in regards to sensor readings, it's true they can be all over the place
as whatevergreen is sending sensors.

You have to appreciate the macOS desktop is quite graphics heavy in the
first place (the Cocao framework) which is why 8ts buttery smooth most of
the time.

I think hd4600 is supposed to scale from 200mhz to 1350mhz so I will see if
I can bring the graphics down a bit without effecting performance.

It might just be a reading error.


Anyway I hardly use my Dells as a hackintosh now as I moved to a t440p
ThinkPad that's quad core and triple SSD with better cooling (the ultra low
voltage dual cores are not great for development)

But I am glad your enjoying it and this pack works well for you.

I will see what can be done in the rest of ssdt to bring down the gpu
baseline

**Comment by coreygreenberg** [comment]
Well whatever packs you collected from others must be hidden deep in the dark web because I searched high and low for the lowdown on making these Dells work as MacBooks and nothing I Googled worked even minimally until I found your stuff. So thanks again for sharing it. 

I can see why you moved on to the Thinkpad. It’s a beast. For my needs the 7240 is perfect, more than enough grunt and a smaller and lighter package. My iPad Pro’s screen has ruined commodity 1366x768 TN panels for me but it’s fine for what I do. 

I’m curious whether your Clover config is overclocking the HD 4400 just a bit, or if Intel Power Gadget is somehow misreading the GPU sensor. I’ve never seen this kind of disparity with any of my other (admittedly desktop) Hacks. Intel Power Gadget always tracks precisely with Activity Monitor. This is the first time I see different GPU frequencies from these two monitors. But this is really the only issue I’m seeing with your Clover package. The e7240 is running Mojave superbly, and everything just works. Sleep/wake, Metal, iMessage, everything. I’ve never had such luck with Hackintoshing any other laptop. Thanks again for sharing your work.

**Comment by coreygreenberg** [comment]
One thing I forgot to mention -- although About This Mac correctly ID's the GPU as Intel HD 4400, Activity Monitor shows it as Intel Iris Pro Graphics. As I said, graphics seem to be working fine, but if Activity Monitor thinks the GPU is a different chipset, maybe that's why it shows different readings than Intel Power Gadget.

**Comment by ameeno** [comment]
Tip 1: both my e7240's are 1080p lcd's

You should grab one of eBay and swap it over if you get a chance, however
1080p Res on 12.5inch can makes things a little too small, but nothing
highdpi scaling won't fix.


Secondly, I use virtualsmc with this pack I believe.

It has improvements over fakesmc+sensors but lacks as many sensors as
fakesmc.

If you feel like trying it out, grab fakesmc+ sensors (you can download it
using clover configurator)


And delete the virtualsmcdriver and I stall the SMC driver, this will give
you more sensors including GPU.


I would advise you backup your efi partition before making any changes, or
perhaps put the modified efi on a usb stick and boot from the usb to see if
changes help you.

**Comment by coreygreenberg** [comment]
I tried subbing FakeSMC.kext for VirtualSMC.kext, and I added all the SMC sensors that Clover Configurator offers when installing FakeSMC. Rebooted without issue, lost the battery menu icon but otherwise all was well. Intel Power Gadget and Activity Monitor showed the same GPU behavior as with VirtualSMC, no changes. HWMonitor didn't seem to show any GPU info, only CPU. 

I've thought about swapping these E7240s for the version with the 1080 screen but honestly these do me just fine for what I want from a laptop. Simple web browsing and text editing is fine on 1366 TN. That E7240 1080 screen is a touchscreen, yes? Does touchscreen work at all with MacOS?

I really appreciate your help. At this point I'm going to chalk it up to bad sensor reading and assume the HD 4400 is not getting overclocked somehow. I'll do some more digging into how the Intel GPU is spoofed in Clover, maybe it's got some holdover code from someone's Iris Pro installation.
